File: app/src/main/java/net/tlalka/android/fiszki/injections/components/SessionComponent.java
package net.tlalka.android.fiszki.injections.components;

import dagger.Subcomponent;
import net.tlalka.android.fiszki.injections.annotations.SessionScope;
import net.tlalka.android.fiszki.injections.modules.ActivityModule;
import net.tlalka.android.fiszki.injections.modules.SessionModule;

@SessionScope
@Subcomponent(modules = SessionModule.class)
public interface SessionComponent {

    ActivityComponent add(ActivityModule activityModule);
}

File: app/src/main/kotlin/net/tlalka/fiszki/core/components/SessionComponent.kt
package net.tlalka.fiszki.core.components

import dagger.Subcomponent
import net.tlalka.fiszki.core.annotations.SessionScope
import net.tlalka.fiszki.core.modules.ActivityModule
import net.tlalka.fiszki.core.modules.SessionModule

@SessionScope
@Subcomponent(modules = [SessionModule::class])
interface SessionComponent {

    fun add(activityModule: ActivityModule): ActivityComponent
}
